### PR TITLE
Add CI for MSVC+Cuda

### PR DESCRIPTION
**Pull request by masterleinad**
Based on top of #6659 that is needed for this new build to pass.

### CODE REVIEW

File: .github/workflows/windows.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+  group: ${ {github.event_name }}-${{ github.workflow }}-${{ github.ref }}
+  cancel-in-progress: ${{github.event_name == 'pull_request'}}
+
+permissions:
+  contents: read
+
+jobs:
+  windows-cuda:
+    # Cuda build on Windows
+    name: Windows Cuda
```
Comment (fnrizzi) [inline .github/workflows/windows.yml:17]: can we make the name consistent with the names of the other CI pipelines?

Comment (fnrizzi) [reply .github/workflows/windows.yml:17]: so that is looks consistent with the other names

File: .github/workflows/windows.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+      shell: bash
+      run: |
+        mkdir build
+        mkdir c:/project
+        cd build
+        cmake -DKokkos_ENABLE_CUDA=ON -DKokkos_ARCH_VOLTA70=ON -DKokkos_ENABLE_TESTS=ON -DKokkos_ENABLE_COMPILE_AS_CMAKE_LANGUAGE=ON ..
+    - name: build library
+      shell: bash
+      run: |
+        cmake --build build --parallel 2 --config Release
```
Comment (fnrizzi) [inline .github/workflows/windows.yml:36]: why not adding both a debug and release build?

**Comment by dalg24** [comment]
Any particular reason for using GitHub workflows rather than AppVeyor for that build?
(Naively asking since the Windows testing is currently on there)

**Comment by masterleinad** [comment]
> Any particular reason for using GitHub workflows rather than AppVeyor for that build?

I managed to do it with `GitHub` workflows and could use an existing action for installing `Cuda` (that caches it). I tried previously with `appveyor`, see https://github.com/kokkos/kokkos/pull/5266 but never really managed to get both builds going the way I wanted to (also it didn't cache `cuda`).
I would consider moving the `appveyor` CI we have right now to `GitHub` as well so that we avoid being split across too many different systems.

File: .github/workflows/windows.yml (hunk 1 above)
Comment (masterleinad) [reply .github/workflows/windows.yml:17]: What do you want it to be? `github-windows-cuda / CI`?

**Review by crtrott — APPROVED** [review]

**Pull request merged**